src/term.rs

```diff
@@ -1485,6 +1624,18 @@ pub mod make {
         };
     }
 
+    /// Array for types implementing `Into<ArrayAttrs>` (for the attributes and `Into<RichTerm> for elements.
+    /// `mk_array!(ArrayAttrs::default(), Term::Num(42)) corresponds to `[42]`. Though the evaluated array may have different attributes.
+    #[macro_export]
```

Comment:
A few questions:
1. Does the `Into<ArrayAttrs>` make sense? For RichTerm it does, because we have all those different nested types (`RichTerm`, `SharedTerm`, `Term`). However I can't think of something interesting that would be `Into<ArrayAttrs>`.
2. It could be useful to have the array attrs optional, by using a different separator, like `;` at the end (a bit like the macro for row types does about the tail).